engine/classes/ElggFile.php

```diff
@@ -233,28 +298,51 @@ public function write($data) {
 	 *
 	 * @return mixed Data or false
 	 */
-	public function read($length, $offset = 0) {
+	public function readOld($length, $offset = 0) {
 		$fs = $this->getFilestore();
 
 		return $fs->read($this->handle, $length, $offset);
 	}
 
+	
 	/**
 	 * Gets the full contents of this file.
 	 *
 	 * @return mixed The file contents.
 	 */
 	public function grabFile() {
+		return $this->getDataStorage()->file($this->getFilename())->getContent();
+	}
+
+	
+	/**
+	 * @todo BC
+	 * 
+	 * Gets the full contents of this file.
+	 *
+	 * @return mixed The file contents.
+	 */
+	public function grabFileOld() {
```

Comment:
All these *Old methods are making me think we should use some good old polymorphism instead:

ElggFileOld vs ElggFileNew classes, all the methods stay the same -- grabFile, read, etc.

Better names would be good, but you get the idea.